sdk/client/client.go

```diff
@@ -189,6 +189,11 @@ func (r *Response) Unmarshal(model interface{}) error {
 		// Trim away a BOM if present
 		respBody = bytes.TrimPrefix(respBody, []byte("\xef\xbb\xbf"))
 
+		// In some cases the respBody is empty, but not nil, so don't attempt to unmarshal this
+		if len(respBody) == 0 {
+			return nil
+		}
```

Comment:
Mind implementing the same fix for XML deserialization too (but _not_ for the bytes deserialization, which'd want that context)?